keps/wg-lts/20200122-kubernetes-yearly-support-period.md

```diff
@@ -0,0 +1,313 @@
+
+There are many unknowns about changing the support windows for a project with as many moving parts as Kubernetes, and keeping the change relatively small (relatively being the important word), gives us the chance to find out what those unknowns are in detail and address them.
+
+### Goals
+
+- Address end-user feedback regarding support time per release
+- Define an annual support policy.
+- Define any prerequisites for the policy.
+- Set acceptance criteria for enacting the change in support policy. This should be supplemented by KEPs and PRs that address the implementation of the policy by the release team and Release Engineering (patch management).
+- Clarify and confirm that the support period is defined based on time (e.g. 1 year) and not based on a specific number of releases (e.g. 3 or 4) given that the release cadence might change in the future.
+
+### Non-Goals
+
+- Define all details of “support”, which is somewhat imprecise in the project already today.
+- Change the release cadence.  It is currently 3 months per release or 4 releases per year.  See “Alternatives” below for related work on release cadence.
+- Establish an “LTS” release process in the sense perhaps known from other projects (e.g.: pick one release every year or two, give patch support on that release for multiple years).
+- Declare that Kubernetes’ dependencies must also have a similar support lifetime per release.
+- Align Kubernetes releases in time with those of its dependencies.
+
+## Proposal
+
+The proposal is to extend the Kubernetes support cadence from 9 months to 14 months.
+The 14 months proposed here is essentially 12 months of support and 2 months of upgrade period.
+NOTE: This follows and formalizes today’s practice of retaining patch branch CI for 6-7 weeks beyond the 9 months intended support phase as a failsafe option if any highly critical CVEs or blocking upgrade issues are observed by slower adopters of newer patch branches.
+
+A visual example today vs. proposed for an adopter of 1.18 in April 2020:
+
+![Visual example of upgrade cadences](./20200122-upgrade-cadence.png)
+
+## Design Details
+
+### Component changes
+
+* Adjust behavior in the kube-apiserver that assumes a component lifetime of ~9 months
+  * Lifetime of the certificate used for kube-apiserver loopback connections (#86552)
+* Expand supported kubelet/apiserver skew to 12 months divided by release cadence minus 1, i.e. currently (12/3)-1=3 releases.
+  * Currently, supported skew (2 versions) was chosen to allow the oldest kubelet to work against newest apiserver
+  * Implementation should include adding a test to cover the added skew variations, have it release informing, watch for any errors, fix them, promote the test to release blocking.  This is already a gap today and can be treated as a project need orthogonal to this KEP.
+* Expand supported kubectl/apiserver skew
```

Comment:
This item also needs feedback from SIG-CLI and SIG-API-Machinery (client libraries)